KeeAgent/KeeAgentExt.cs

```diff
@@ -815,35 +815,29 @@ private void PageantAgent_KeyRemoved(object sender, SshKeyEventArgs e)
       RemoveKey(e.Key);
     }
 
+    private object unlockOnActivitySyncRoot = new object(); // used in PageantAgent_MessageReceived as a sync root
+
     private void PageantAgent_MessageReceived(object sender, Agent.MessageReceivedEventArgs e)
     {
+      if (!Options.UnlockOnActivity) {
+        return; // unlock on activity isn't enabled, nothing to do here
+      }
+
       var mainWindow = pluginHost.MainWindow;
+      if (!mainWindow.DocumentManager.Documents.Any(document => mainWindow.IsFileLocked(document))) {
+        return; // none of the documents are locked, nothing to do here
+      }
 
-      var thread = new Thread(
-        delegate()
-        {
-          mainWindow.Invoke(
-            (MethodInvoker)delegate()
-          {
-            // don't do anything - we are just seeing if the thread is blocked
-          });
-        });
-      thread.Name = "Check";
-      thread.Start();
-      // only try to unlock databases if main thread is not blocked

```

Comment:
Does removing this extra thread check cause a regression of https://github.com/dlech/KeeAgent/issues/19?

Reply:
@dlech , my bad.
I thought this check had something to do with the problem I was dealing with and I wasn't smart enough to have a look at git blame to find out why this check was added in the first place.
Fixed.